### DISCUSSION TITLE
How do I get the Output for the current Command

### DISCUSSION THREAD

**Discussion by finn-matti** [symfony/symfony#42488] [Q&A] [answered]
I have a quick question: How do I receive the current output from the current Command?

I know that I have them available to me in all of the methods Symphony's Command class implements, but what if I want to have my own methods? Do I really have to add a parameter to my own methods?

Example:

```php

// ...
use Symfony\Component\Console\Command\Command;
// ...

class MyCommand extends Command

// ...

{
    protected function logAndPrint($message, OutputInterface $output)
    {
           $this->log($message);
           $output->writeln($message);
   }
}

// ...
```

Do I really have to handle the `$output` like this? I couldn't find a `$this->getOutput()` method or similar.

Thanks in advance!

**Comment by stof** [comment, marked as answer]
There is no `getOutput` in Symfony because the Command receives the output object in its `execute()` method, not as part of its instantiation.

**Reply by javiereguiluz** [reply]
Christophe is right. You must pass the `$output` variable around ... or store it in a private property of the command and do `$this->output` (that's what I usually do to avoid passing the variable).

**Reply by finn-matti** [reply]
Thanks for the clarification. Then I'll use a property. 👍
